src/AppBundle/Controller/MembershipController.php

```diff
@@ -102,28 +99,16 @@ public function donateAction(Request $request): Response
      * @Route("/inscription/centre-interets", name="app_membership_pin_interests")
      * @Method("GET|POST")
      */
-    public function pinInterestsAction(Request $request): Response
+    public function pinInterestsAction(Request $request, RegisteringAdherent $registering): Response
     {
-        $membershipUtils = $this->get('app.membership_utils');
-
-        if (!$id = $membershipUtils->getNewAdherentId()) {
-            throw $this->createNotFoundException('The adherent has not been successfully redirected from the registration page.');
-        }
-
-        $manager = $this->getDoctrine()->getManager();
-
-        if (!$adherent = $manager->getRepository(Adherent::class)->find($id)) {
-            throw $this->createNotFoundException('New adherent id not found.');
-        }
-
-        $form = $this->createForm(AdherentInterestsFormType::class, $adherent)
+        $form = $this->createForm(AdherentInterestsFormType::class, $registering->getAdherent())
             ->add('pass', SubmitType::class)
             ->add('submit', SubmitType::class)
         ;
 
         if ($form->handleRequest($request)->isSubmitted()) {
             if ($form->get('submit')->isClicked() && $form->isValid()) {
-                $manager->flush();
+                $this->getDoctrine()->getManager()->flush();
```

Comment:
`$this->get('doctrine.orm.default_entity_manager')` and because it's too long create an alias `em`